### PR TITLE
test: tests to show predicate simplification on chunks

### PR DESCRIPTION
**Pull request by alamb**
Closes https://github.com/influxdata/influxdb_iox/issues/3601

Demonstrates that `_measurement` predicates are simplified prior to being passed to chunks (for metadata analysis)

### CODE REVIEW

File: query/src/frontend/influxrpc.rs [hunk 1]
```diff
@@ -1953,90 +1953,84 @@ mod tests {
...
-    where
-        F: FnOnce(&TestDatabase, InfluxRpcPredicate) + Send,
-    {
-        let chunk0 = Arc::new(
-            TestChunk::new("h2o")
-                .with_id(0)
-                .with_tag_column("foo")
-                .with_time_column(),
-        );
+    /// Runs f() which is expected to call a gRPC planning function
```
Comment (alamb) [inline query/src/frontend/influxrpc.rs:2029]: I removed the generics mostly to make the code easier to read -- the passed in function gets called 3 times

File: query/src/frontend/influxrpc.rs [hunk 2]
```diff
@@ -2053,22 +2047,73 @@ mod tests {
...
             .build();
 
+        // verify that the predicate was rewritten to `foo = 'bar'`
+        let expr = col("foo").eq(lit("bar"));
+        let expected_predicate = PredicateBuilder::new().add_expr(expr).build();
+
+        run_test_with_predicate(&f, silly_predicate, expected_predicate).await;
+
+        // ------------- Test 2 ----------------
+        // Validate that _measurement predicates are translated
```
Comment (alamb) [inline query/src/frontend/influxrpc.rs:2057]: here are the new tests inspired by https://github.com/influxdata/influxdb_iox/issues/3601

File: query/src/frontend/influxrpc.rs [hunk 3]
```diff
@@ -2053,22 +2047,73 @@ mod tests {
...
+            .add_expr(
+                col("_measurement")
+                    .eq(lit("foo"))
+                    .or(col("_measurement").eq(lit("h2o")))
+                    .and(col("time").gt(lit(5))),
+            )
+            .build();
+
+        // verify that the predicate was rewritten to time > 5
+        let expr = col("time").gt(lit(5));
```
Comment (alamb) [inline query/src/frontend/influxrpc.rs:2087]: I am quite pleased with these rewrites;  they are pretty sophisticated  :bowtie:

File: query/src/frontend/influxrpc.rs (hunk 1 above)
Comment (tustvold) [reply query/src/frontend/influxrpc.rs:2029]: I personally find this harder to read, but perhaps I have Stockholm syndrome :laughing: 

That being said, I'm not sure what the `D` is doing...

**Review by tustvold — APPROVED** [review]
:man_dancing:

File: query/src/frontend/influxrpc.rs (hunk 1 above)
Comment (alamb) [reply query/src/frontend/influxrpc.rs:2029]: Yeah, I can't remember why I had to make the `D` either 🤔 

I'll make `&(dyn Fn(&TestDatabase, InfluxRpcPredicate) + Send + Sync)` a type alias and add some comments and hopefully that will make it ok-ish readability wise

**Pull request merged**